components/s3_bucket/main.tf

```diff
@@ -62,6 +68,26 @@ resource "aws_s3_bucket" "bucket" {
     enabled = var.versioning || var.replication_target != null
   }
 
+  dynamic "lifecycle_rule" {
```

Comment:
Because of how this `aws_s3_bucket` resource was built by Hashicorp/AWS, we have to configure this lifecycle rule as a dynamic "property" on the `aws_s3_bucket` (rather than as it's own optional resource that we could point to the bucket it's being attached to, like we do with a lot of other things).

Terraform's dynamic blocks are described in more detail [here](https://www.terraform.io/docs/language/expressions/dynamic-blocks.html).